docs/AddNewOp.md

```diff
@@ -48,11 +60,35 @@ rank inference at the very least (adding right amount of dimensions to the outpu
 ### Example to Follow
 [PR 1959](https://github.com/onnx/onnx/pull/1959) is a good example to follow.
 
-## Step 3: PR Review by Operators SIG
+## Step 3: PR Review by Operators SIG <a name="step3_new_operator_or_function"></a>
 The [Operators SIG](https://github.com/onnx/sigs/tree/master/operators) is responsible for the operators/functions in the ONNX specification. The SIG regularly meets and reviews PRs.
 
 ### Sign-off
 At least two sign-off from the Operators SIG [contributors](https://github.com/onnx/onnx/tree/master/community#community-roles).
 
-## Step 4: ONNX release
+## Step 4: ONNX release <a name="step4_new_operator_or_function"></a>
 Once the PR is reviewed and signed off by the Operators SIG, it will be merged. Your new operator/function will be part of the master branch and available to anyone building from source. These are not official releases. ONNX periodically releases official new versions that are a snapshot of the master branch. Your new operator/function will be part of that release.
+
+# Removing operator or function <a name="removing_operator_or_function"></a>
+There are a lot of reasons for removing existing ONNX operator or function, such us being replaced with different operator or can be decomposed by a set of other operators. This document describes the criteria of removing an existing ONNX operator from the standard.
+
+## Removing operator <a name="removing_operator"></a>
+Any operator in ONNX was added because it was required by a model and/or framework. In order to deprecate such an operator we need to do the following.
+* Operator can’t be deprecated unless there is a replacement.
+    * Replacement can be a more general operator that supersedes the old one.
+    * Or a set of primitive operators that together can implement the same functionality and behavior of the deprecated operator (Function).
+* If the deprecated operator can be decomposed by existing operators then it must be converted to a function.
+* If replacement isn’t in ONNX standard yet, then add the replacement operator or set of operators first.
+* No grace period is needed for deprecated operators.
+
+## Removing function <a name="removing_function"></a>
+Function, by definition, is composed of ONNX primitives; however, function could have been accelerated by framework or runtime that support ONNX. So, removing function is not recommended, with the exception of adding another single function which supersede its functionality.
+
+## Document removing operator or function <a name="document_removing_operator_or_function"></a>
+To make sure everyone is aware of the deprecation, the following need to happen:
+* Any removed operator or function from ONNX need to be mentioned in the release note.
+* Their old documentation needs to be updated to show the new replacement and the mapping between the old to the new.
+    * Only `def.cc` need to be remove, `old.cc` will remain.
+    * `old.cc` need to be updated with the mapping to the replacement.
```

Comment:
I guess the new replacement description will be in addition to the old definition, right?

Reply:
Yes.